File: CursoJava/Exer/src/Outros/ListaAlgoritimo/Algoritimo3/Exer04.java
/*
4) Escrever um algoritmo que leia uma quantidade desconhecida de números e conte quantos deles
estão nos seguintes intervalos: [0-25], [26-50], [51-75] e [76-100]. A entrada de dados deve
terminar quando for lido um número negativo.
 */

package Outros.ListaAlgoritimo.Algoritimo3;

import java.util.Scanner;

public class Exer04 {
    public static void main(String[] args) {
        Scanner sc = new Scanner(System.in);

        int num, cont1 = 0, cont2 = 0, cont3 = 0, cont4 = 0;

        do {
            System.out.print("Digite um número: ");
            num = sc.nextInt();

            if (num >= 0 && num <= 25) {
                cont1++;
            } else if (num >= 26 && num <= 50) {
                cont2++;
            } else if (num >= 51 && num <= 75) {
                cont3++;
            } else if (num >= 76 && num <= 100) {
                cont4++;
            }
        } while (num >= 0);

        System.out.println("Quantidade de números no intervalo [0-25]: " + cont1);
        System.out.println("Quantidade de números no intervalo [26-50]: " + cont2);
        System.out.println("Quantidade de números no intervalo [51-75]: " + cont3);
        System.out.println("Quantidade de números no intervalo [76-100]: " + cont4);

        sc.close();
    }
}

File: CursoLuizOtavio/CursoPython3/exer/lista_algoritimos/algoritimo3/exer04.py
"""
4) Escrever um algoritmo que leia uma quantidade desconhecida de números e conte quantos deles
estão nos seguintes intervalos: [0-25], [26-50], [51-75] e [76-100]. A entrada de dados deve
terminar quando for lido um número negativo.
"""

count_0_25 = 0
count_26_50 = 0
count_51_75 = 0
count_76_100 = 0

while True:
    num = int(input("Informe um número (negativo para sair): "))

    if num < 0:
        break

    if 0 <= num <= 25:
        count_0_25 += 1
    elif 26 <= num <= 50:
        count_26_50 += 1
    elif 51 <= num <= 75:
        count_51_75 += 1
    elif 76 <= num <= 100:
        count_76_100 += 1

print(f"Quantidade de números no intervalo [0-25]: {count_0_25}")
print(f"Quantidade de números no intervalo [26-50]: {count_26_50}")
print(f"Quantidade de números no intervalo [51-75]: {count_51_75}")
print(f"Quantidade de números no intervalo [76-100]: {count_76_100}")
